### DISCUSSION TITLE
How to execute binary script from dependency in my project?

### DISCUSSION THREAD

**Discussion by Chaz0r** [yarnpkg/berry#5398] [Q&A] [answered]
Hey guys,
probably a pretty simple question for you but every solution I read about didn't work. So I'm not sure if I am doing something wrong or if there is bug or something else.

First of all: I'm using yarn 2 with pnp enabled.

I created a simple library. Lets name it **_@test/mylib_**. In the package.json of my lib I added a bin script:
```
//part of package.json 
 "bin": {
    "test123": "./scripts/testtest.js",
  },
```

As you can see there is a simple script and this script simply does not do something else beside a console.log("WORKS").
In the terminal of my lib I can execute `yarn run test123` and I can see the console output. So that works.

Now I have another project. This project includes my library as a dependency. What I want is to execute the test123 script of my library in the terminal of my project. 

By executing `yarn info` I can see that the library is correctly included in the latest version 0.0.22. I can also see that it has an exposed binary called `test123`

So as I understood I now should execute the following statement and it should work:
`yarn dlx @test/mylib test123`
But it doesn't.

What I get is an internal error **Binary not found (mylib) for root-workspace-0b6124@workspace.** 

How can I execute the binary script of my library outside of my library?

Thanks in advance!

**Comment by merceyz** [comment, marked as answer]
Since the binary name doesn't match the package name (`mylib`) you need to specify which one you want to run:
```sh
yarn dlx -p @test/mylib test123
```

**Reply by Chaz0r** [reply]
Jesus, the -p was the missing part. Thanks!
